src/bitcoin-cli.cpp

```diff
@@ -316,7 +316,15 @@ static UniValue CallRPC(BaseRequestHandler *rh, const std::string& strMethod, co
 
     // Synchronously look up hostname
     raii_evhttp_connection evcon = obtain_evhttp_connection_base(base.get(), host, port);
-    evhttp_connection_set_timeout(evcon.get(), gArgs.GetArg("-rpcclienttimeout", DEFAULT_HTTP_CLIENT_TIMEOUT));
+
+    // Set connection timeout
+    int timeout = gArgs.GetArg("-rpcclienttimeout", DEFAULT_HTTP_CLIENT_TIMEOUT);
+    if (timeout > 0) {
+        evhttp_connection_set_timeout(evcon.get(), timeout);
+    } else {
+        const int YEAR_IN_SECONDS = 31556952; // Average length of year in Gregorian calendar
```

Comment:
maybe add a comment that this is a workaround for libevent-http having no way to set an infinite timeout

Reply:
done